### ISSUE TITLE
Grand Central M4 (SAMD51) - pwmio not functioning and no error being rasied

### ISSUE THREAD

**Issue by AdamCummick** [adafruit/circuitpython#3781]
Running this code causes D19 to idle high and not oscillate. Removing D30 from pins_temp corrects the issue. No errors are flagged. D30 and D19 share TC4_WO1, but have different TCC peripherals.

```py
import board
import time
import pwmio

pins_temp = [ board.D30, board.D19]
pwm_pins = []
for pin in pins_temp:
    p = pwmio.PWMOut(pin, frequency=10, duty_cycle=0x7fff, variable_frequency=True)
    pwm_pins.append(p)

while True:
    print(10)
    for pin in pwm_pins:
        pin.frequency = 10
    time.sleep(3)
    print(20)
    for pin in pwm_pins:  
        pin.frequency = 20
    time.sleep(3)
```

**Comment by tannewt** [comment]
What version of CircuitPython are you using?

**Comment by AdamCummick** [comment]
I'm running 6.0.0

**Comment by dhalbert** [comment]
Reproduced in 5.3.1 and later. The problem does not happen if `.frequency` is not assigned, and we just use the original frequency.

**Comment by dhalbert** [comment]
@AdamCummick This will be fixed by #3960.
